### PR TITLE
update snap package

### PR DESCRIPTION
**Pull request by siennathesane**
update snapcraft wrapper to use core22 as the current lts and all of the build & deploy deps for snap

relates to #27014

### CODE REVIEW

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -44,12 +44,13 @@ parts:
     source: .
     stage-packages: [ git, sqlite3, openssh-client ]
     build-packages: [ git, libpam0g-dev, libsqlite3-dev, build-essential]
-    build-snaps: [ go, node/18/stable ]
+    build-snaps: [ go/latest/stable, node/18/stable ]
```
Comment (silverwind) [inline snap/snapcraft.yaml]:
```suggestion
    build-snaps: [ go/1.21/stable, node/18/stable ]
```

given that we lock go every where else, we should do here as well.

File: .gitignore [hunk 2]
```diff
@@ -103,6 +103,9 @@ prime/
 *.snap-build
 *_source.tar.bz2
 .DS_Store
+gitea_amd64*.txt
+gitea_arm64*.txt
+gitea_armhf*.txt
```
Comment (silverwind) [inline .gitignore]: Move to `Snapcraft` header on line 97 and change to `/gitea_*.txt` to match exactly (assuming they generate in root).

Comment (siennathesane) [reply .gitignore]: Updated

**Review by techknowlogick — APPROVED** [review]
Thanks for this PR, and for the one to tea too :)

**Comment by silverwind** [comment]
Should we upgrade to node 20? @techknowlogick I noticed you reverted to node 18, any reason?

**Comment by techknowlogick** [comment]
@silverwind yeah, the snap builder doesn't like 20 (for our project at least). I've found that I have to keep it a few versions behind. It's hard to debug because it only pulls the config from the main branch, and it builds fine locally just not in the remote runner.
Maybe we switch from the remote runner to Actions, but from past attempts that's tricky building for multi-arch snaps.

**Review by silverwind — APPROVED** [review]

**Pull request merged**